### PR TITLE
fix: Fix AMVF KF update

### PR DESCRIPTION
**Pull request by andiwand**
Taking a block from the weight matrix is only possible if time component is identity

### CODE REVIEW

File: Core/include/Acts/Vertexing/KalmanVertexTrackUpdater.ipp [hunk 1]
```diff
@@ -19,18 +19,13 @@ void Acts::KalmanVertexTrackUpdater::update(TrackAtVertex<input_track_t>& track,
   // Get the linearized track
   const LinearizedTrack& linTrack = track.linearizedState;
 
-  // Check if linearized state exists
-  if (linTrack.covarianceAtPCA.determinant() == 0.) {
-    // Track has no linearized state, returning w/o update
-    return;
-  }
```
Comment (felix-russo) [inline Core/include/Acts/Vertexing/KalmanVertexTrackUpdater.ipp:26]: Should we maybe check `if (track.isLinearized)` here? If the track is not linearized I think that `posJack` etc. are just zero matrices (but I might be wrong).

Comment (andiwand) [reply Core/include/Acts/Vertexing/KalmanVertexTrackUpdater.ipp:26]: that would definitely make more sense than checking the `det`. I just noticed that we don't check that in the other KF updater and thought it makes sense to align that

Comment (felix-russo) [reply Core/include/Acts/Vertexing/KalmanVertexTrackUpdater.ipp:26]: I think we linearize the track before we use it to update the vertex 
image

**Review by felix-russo — DISMISSED** [review]

**Review by felix-russo — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6070915)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847894) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847893) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847892) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847891) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847890) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847889) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847888) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847887) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847886) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847885) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847884) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847883) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847882) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847881) |
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847880) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847879) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31847878) |

**Comment by andiwand** [comment]
looks like this is breaking the athena tests https://github.com/acts-project/acts/pull/2381#issuecomment-1685082511

interestingly it did not change our physmon outputs

not sure how big the difference in athena is

**Pull request merged**